pytorch_lightning/utilities/data.py

```diff
@@ -47,6 +48,11 @@ def _extract_batch_size(batch: BType) -> Generator[int, None, None]:
         if isinstance(batch, Mapping):
             batch = batch.values()
 
+        for sample in batch:
+            yield from _extract_batch_size(sample)
+    elif is_dataclass(batch):
+        batch = [getattr(batch, field) for field in batch.__dataclass_fields__]
```

Comment:
```suggestion
        batch = [getattr(batch, field.name) for field in dataclasses.fields(batch)]
```
I think this works too. Just avoiding private method calls.

Reply:
I agree, has already been suggested on Slack. Just haven't found time to fix